### ISSUE TITLE
generateProof in zokrates-js

### ISSUE THREAD

**Issue by rayyan808** [Zokrates/ZoKrates#922]
## Description

Whilst attempting to generate a proof, with a valid witness, proving key and program passed as arguments I encounter an error : Runtime unreachable. I've already tested my inputs. Is this maybe related to the previous issue I reported: #912 ? Though, these binaries are generated through zokrates-js to workaround that bug, utilizing `zkProvider.compile`

On a side-note: It is definitely not bad state management that is causing this, as the console.log logs prove the validity of the outputs.
## Environment

- Compiler version: 0.7.3
- Zokrates-JS: 1.0.32
- Operating system: Ubuntu 16

## Steps to Reproduce

```
initialize().then((zkProvider) => {
       console.log("MembershipTest(" + this.state.secretKey + ", " + merkleRoot + ", " + dirSelector + "," + siblingNodes);
      const {witness, computationResult} = zkProvider.computeWitness(this.state.membershipGenerator, [this.state.secretKey, merkleRoot, dirSelector, siblingNodes]);
      console.log("Your witness result: " + witness);
      console.log("Computation Output: " + computationResult);
      this.setState({generatedWitness: witness});
      console.log("End of Witness Conduct \n");
      }).catch((err) => {
          console.log("Error caught during witness computation: " + err);
          this.setState({generatedWitness: null});
      }).finally(() => {
        console.log("Async Witness Op concluded.");
                /* Only after witness computation, begin proof generation*/
                if(this.state.generatedWitness != null){
                        initialize().then((zkProvider) => {
                        console.log("Conducting Proof Generation")
                        console.log("Your program: " + this.state.membershipGenerator.program);
                        let localProof = zkProvider.generateProof(this.state.membershipGenerator.program, this.state.generatedWitness, this.state.provingKey);
                        console.log("your local proof: " + localProof);
                        this.setState({proof:localProof}); 
                        }).catch((err) => {
                        console.log("Error during Proof Generation: " + err);
                        }).finally( () => {
                        console.log("Conduct Finished.")
                      });
    } else {
      console.log("You provided an invalid combination of secretKey and VoteKey, thus the proof could not be generated.");
    }
  });
```

The ERROR:
```
submitVote.js:257 Error during Proof Generation: RuntimeError: unreachable
```

**Comment by dark64** [comment]
Judging by the error, it seems to be related to the proving key as the error thrown (`Custom { kind: InvalidData, error: NotOnCurve }`) comes from the deserialization of the proving key. How and where do you set `this.state.provingKey`?
Also, you don't have to initialize twice, `zkProvider` should be still in scope in your `finally` block.

**Comment by rayyan808** [comment]
I initially had one initialization, but split it into two whilst debugging. The proving key is obtained the same way that the .zok file is loaded and then compiled from: 

```
getProvingKey = (event) => {
  console.log("Get Proving Key caled");
  const file = event.target.files[0];
  //console.log('files: ' + file);
  const reader = new FileReader();
  reader.readAsText(file);
  reader.addEventListener('load', (e) => {
    const buffer = reader.result;
    console.log("Data loaded successfully");
    this.setState({provingKey: buffer.toString()/*CHANGED*/});
  });
}
```

**Comment by dark64** [comment]
Proving key is serialized to a binary format, not a string. Try this:
```js
getProvingKey = (event) => {
  console.log("Get Proving Key caled");
  const file = event.target.files[0];
  //console.log('files: ' + file);
  const reader = new FileReader();
  reader.readAsArrayBuffer(file);
  reader.addEventListener('load', (e) => {
    const buffer = new Uint8Array(reader.result);
    console.log("Data loaded successfully");
    this.setState({provingKey: buffer});
  });
}
```

**Comment by rayyan808** [comment]
Thank you, this solved the issue!
